src/main/java/io/nats/client/impl/NatsConnectionWriter.java

```diff
@@ -75,6 +75,27 @@ class NatsConnectionWriter implements Runnable {
         reconnectBufferSize = options.getReconnectBufferSize();
     }
 
+    NatsConnectionWriter(NatsConnectionWriter sourceWriter) {

```

Comment:
Added to replace an existing writer but keeps it's relevant state including the message queue.